mq/plugins/complianceitems.py

```diff
@@ -58,17 +58,21 @@ def onMessage(self, message, metadata):
             The complianceitems plugin is called when an event
             is posted with a doctype 'complianceitems'.
             Compliance items are stored in the complianceitems
-            index, with doctype last_known_state
+            index, with a type of last_known_state. The doc_type
+            is overwritten with _doc.
         """
         if not self.validate(message['details']):
             logger.error('Invalid format for complianceitem {0}'.format(message))
             return (None, None)
+        if 'type' not in message:
```

Comment:
I don't think we want to check if `type` already exists, I think we want to automatically set that value no matter what the incoming message looks like.

Reply:
I've set this to be set everytime. I had thought type was in the message already but then realized that the place where it occurs looks for it specifically several keys down, not at the top.